plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```
@@ -994,10 +1007,16 @@ private Map<String, String> getEmptyTableProperties(ConnectorTableMetadata table
 
         // Avro specific properties
         String avroSchemaUrl = getAvroSchemaUrl(tableMetadata.getProperties());
+        String avroSchemaLiteral = getAvroSchemaLiteral(tableMetadata.getProperties());
         if (avroSchemaUrl != null) {
+            checkState(avroSchemaLiteral == null, "avro_schema_url and avro_schema_literal cannot both be set");
```

This should throw `TrinoException(INVALID_TABLE_PROPERTY)`